bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -252,6 +252,26 @@ type SecretFileSource struct {
 	Key string `json:"key"`
 }
 
+// PasswdSource is a union of all possible external source types for passwd data.
+// Only one field may be populated in any given instance. Developers adding new
+// sources of data for target systems should add them here.
+type PasswdSource struct {
+	// Secret represents a secret that should populate this file.
+	Secret SecretPasswdSource `json:"secret"`
```

Comment:
Q: should this be a pointer omit empty?
@JoelSpeed @vincepri opinions?

Reply:
In this case no, it shouldn't. These fields are all required, while the `PasswdFrom` field under User is optional and omitempty